compiler/rustc_resolve/src/lib.rs

```diff
@@ -1613,10 +1613,11 @@ impl<'a> Resolver<'a> {
     ) -> SmallVec<[LocalDefId; 1]> {
         let mut import_ids = smallvec![];
         while let NameBindingKind::Import { import, binding, .. } = kind {
-            let id = self.local_def_id(import.id);
-            self.maybe_unused_trait_imports.insert(id);
+            if let Some(id) = import.id().map(|id| self.local_def_id(id)) {
```

Comment:
Nit: could you put the call to `local_def_id` as a simple statement inside the block?